backend/telemetry_core/src/state/chain.rs

```diff
@@ -35,6 +36,298 @@ id_type! {
 pub type Label = Box<str>;
 
 const STALE_TIMEOUT: u64 = 2 * 60 * 1000; // 2 minutes
+const STATS_UPDATE_INTERVAL: Duration = Duration::from_secs(5);
+
+#[derive(Default)]
+struct Counter<K> {
+    map: HashMap<K, u64>,
+    empty: u64,
+}
```

Comment:
Could we have a comment for this? Is `empty` the "default" value? Perhaps this `Counter` struct should be split out into its own file, so it's super obvious what the interface it exposes is (and the internals can't be poked at)?